src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/RuntimeAsyncRewriter.cs

```diff
@@ -0,0 +1,139 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+
+namespace Microsoft.CodeAnalysis.CSharp;
+
+internal class RuntimeAsyncRewriter : BoundTreeRewriterWithStackGuard
+{
+    public static BoundStatement Rewrite(
+        BoundStatement node,
+        MethodSymbol method,
+        TypeCompilationState compilationState,
+        BindingDiagnosticBag diagnostics)
+    {
```

Comment:
It looks like this code can be reached with non-async methods. Let's bail out early or avoid calling entirely (then we can assert we're only getting async methods here)